### PR TITLE
[Cleanup] Switching ComboBoxes To Async && Test Creation For ComboBox

### PR DESCRIPTION
**Pull request by Civolilah**
@turbo124 The conversion of all `DebouncedComboboxes` to `ComboboxAsync` is finished across the whole app. This includes these selectors:

- Design Selector
- Expense Category Selector
- Bank Account Selector
- Task Status Selector
- Adjustment for User Selector
- Applying User Selector across the app
- Vendor Selector
- Switching comboboxes for Schedule pages
- Group settings selector

Apart from just switching Comboboxes to Async, I improved the structure of the code wherever possible and made the selectors more globalized.

Let me know your thoughts.

### CODE REVIEW

**Comment by Civolilah** [comment]
@beganovich @turbo124 Just to confirm, the tests for the Combobox component are implemented. The implementation of the new Combobox component across the app was completed earlier, so the PR is ready for reviews. Let me know your thoughts.

**Comment by turbo124** [comment]
@Civolilah 

Looks great, just one small but related issue is to how the combo box is being supported for clients in the schedules page.

image


On selection, the client is not popped off the list. it also appears that we are not invalidating the /clients route correctly after adding a new client, as the newly added client is not visible in this list afterwards.

otherwise looks good for merge.

**Comment by Civolilah** [comment]
> 

@turbo124 Really great catch! Both of those bugs are fixed now. Let me know your thoughts.

File: src/pages/settings/schedules/common/components/EmailStatement.tsx [hunk 1]
```diff
@@ -48,11 +48,20 @@ export function EmailStatement(props: Props) {
 
   const [selectedClients, setSelectedClients] = useState<Client[]>([]);
 
+  const handleChangeParameters = (clients: Client[]) => {
```
Comment (Civolilah) [inline src/pages/settings/schedules/common/components/EmailStatement.tsx:51]: @beganovich After David reported bugs related to Schedules during testing, I found an additional one. So, this is just a quick fix for that one.

File: src/components/TestingRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Navigate, Outlet } from 'react-router-dom';
+
+export function TestingRoute() {
+  const isTestEnv = import.meta.env.VITE_IS_TEST === 'true';
+
+  return isTestEnv ? <Outlet /> : <Navigate to="/logout" />;
+}
```
Comment (beganovich) [inline src/components/TestingRoute.tsx:17]: Can this be inlined in the <Routes> part? One less file, if  possible.

File: src/pages/clients/create/Create.tsx [hunk 3]
```diff
@@ -94,6 +97,16 @@ export default function Create() {
...
 
+        queryClient.invalidateQueries('/api/v1/clients');
+
+        window.dispatchEvent(
+          new CustomEvent('invalidate.combobox.queries', {
+            detail: {
+              url: endpoint('/api/v1/clients'),
+            },
+          })
+        );
```
Comment (beganovich) [inline src/pages/clients/create/Create.tsx:108]: I don't think this is a thing anymore?

File: src/components/TestingRoute.tsx (hunk 2 above)
Comment (Civolilah) [reply src/components/TestingRoute.tsx:17]: Sure, done!

File: src/pages/clients/create/Create.tsx (hunk 3 above)
Comment (Civolilah) [reply src/pages/clients/create/Create.tsx:108]: Yeah u are right, I missed that we do not need this anymore. Removed!

**Comment by beganovich** [comment]
Ping @Civolilah for merge conflicts.

**Comment by Civolilah** [comment]
> Ping @Civolilah for merge conflicts.

@beganovich Done!

**Pull request merged**